### PR TITLE
Tecton abilities pass #1

### LINKED ISSUE #20: Tecton abilities pass #1

### PR DESCRIPTION
**Pull request by SheerSt**
Closes #20

### CODE REVIEW

File: core/src/main/java/io/github/fourlastor/game/demo/state/machine/AttackMelee.java [hunk 1]
```diff
@@ -85,21 +96,60 @@ public Action setupAttackAnimation(float distance, float rotationDegrees) {
...
+                break;
+            }
+        }
+        if (ySort != null) {
+            targetTile.actor = new TileOnMap(textureAtlas.findRegion("tiles/white"));
+            targetTile.actor.setPosition(oldActor.getX(), oldActor.getY());
+            ySort.addActor(targetTile.actor);
+            oldActor.remove();
+            ySort.sortChildren();
+        }
```
Comment (fourlastor) [inline core/src/main/java/io/github/fourlastor/game/demo/state/machine/AttackMelee.java]: I think there's a bug in here - the tile should become non-solid

you don't need to remove the old/add a new actor - you can set a change the image on the previous one with `actor.setDrawable(new TextureRegionDrawable(textureRegion))`

Comment (SheerSt) [reply core/src/main/java/io/github/fourlastor/game/demo/state/machine/AttackMelee.java]: I will try that. How do you want YSort to be interacted with?

Comment (fourlastor) [reply core/src/main/java/io/github/fourlastor/game/demo/state/machine/AttackMelee.java]: I think if you don't remove/add it you are already good without touching it

Comment (SheerSt) [reply core/src/main/java/io/github/fourlastor/game/demo/state/machine/AttackMelee.java]: oic

File: core/src/main/java/io/github/fourlastor/game/demo/state/map/MapGraph.java [hunk 2]
```diff
@@ -10,26 +10,28 @@
...
 import java.util.Objects;
 import java.util.Set;
-import java.util.stream.Collectors;
 
 public class MapGraph implements IndexedGraph<Tile> {
 
     private final Heuristic<Tile> heuristic = new TileHeuristic();
-    private final IntObjectMap<Tile> tiles;
-    private final List<Tile> indexed;
+    private final IntObjectMap<Tile> tiles; // (sheerst) Question: what does the index here represent?
```
Comment (fourlastor) [inline core/src/main/java/io/github/fourlastor/game/demo/state/map/MapGraph.java:33]: it's a [compressed](https://github.com/fourlastor-jams/into-the-wip/blob/demo-3/core/src/main/java/io/github/fourlastor/game/coordinates/Packer.java#L7-L9) version of the coordinates

**Comment by SheerSt** [comment]
Can be merged.

**Pull request merged**